Tasks/DockerV2/dockerlifecycle.ts

```diff
@@ -11,33 +11,30 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
     let containerName = tl.getInput("container", true);
     let commandArguments = dockerCommandUtils.getCommandArguments(tl.getInput("arguments", false));
 
+    // Map between specified container names and their actual ID
+    // Allow custom containers names through
+    let container = containerName;
     let containerMap = tl.getVariable("agent.containermapping");
-    if (!containerMap) {
-        throw new Error(tl.loc('ContainerNotFound', containerName));
-    }
-
-    let map: any;
-    try {
-        map = JSON.parse(containerMap);
-    } catch (ex) {
-        throw new Error(tl.loc('ContainerNotFound', containerName));
-    }
-
-    if (!map[containerName]) {
-        throw new Error(tl.loc('ContainerNotFound', containerName));
-    }
-    let containerId = map[containerName].id;
-    if (!containerId) {
-        throw new Error(tl.loc('ContainerNotFound', containerName));
+    if (containerMap) {
+        try {
+            let map = JSON.parse(containerMap);
+            if (map[containerName]) {
```

Comment:
Nit, can combine `.id` check in one line.

Reply:
Yes!